.github/workflows/release.yml

```diff
@@ -0,0 +1,127 @@
+name: Release
+
+run-name: Release ${{ github.ref_name }}
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+permissions: {}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    # see https://docs.github.com/en/actions/how-tos/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token
+    permissions:
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: '3.3'
+        bundler-cache: false
+    
+    - name: Install dependencies
+      run: bundle install --jobs 4 --retry 3
+    
+    - name: Extract version from tag
+      id: version
+      run: echo "VERSION=${GITHUB_REF#refs/tags/v}" >> $GITHUB_OUTPUT
+    
+    - name: Verify version matches code
+      run: |
+        TAG_VERSION="${{ steps.version.outputs.VERSION }}"
+        CODE_VERSION=$(bundle exec ruby -r ./lib/cyclonedx/ruby/version.rb -e 'puts Cyclonedx::Ruby::VERSION')
+        echo "Tag version: $TAG_VERSION"
+        echo "Code version: $CODE_VERSION"
+        if [ "$TAG_VERSION" != "$CODE_VERSION" ]; then
+          echo "::error::Version mismatch! Tag is v$TAG_VERSION but code version is $CODE_VERSION"
+          exit 1
+        fi
+    
+    - name: Determine if prerelease
+      id: prerelease
+      run: |
+        VERSION="${{ steps.version.outputs.VERSION }}"
+        # A version is a prerelease if it contains anything after the patch version
+        # e.g., 1.2.3-alpha.1, 1.2.3.rc1, 1.2.3-beta are all prereleases
+        # but 1.2.3 is a stable release
+        if [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          echo "PRERELEASE=false" >> $GITHUB_OUTPUT
+          echo "Detected stable release: $VERSION"
+        else
+          echo "PRERELEASE=true" >> $GITHUB_OUTPUT
+          echo "Detected prerelease: $VERSION"
+        fi
+    
+    - name: Run tests
+      run: bundle exec rake test
+    
+    - name: Build gem
+      run: bundle exec rake build
+    
+    - name: Generate checksum
+      run: bundle exec rake build:checksum
+    
+    - name: List build artifacts
+      run: ls -lh pkg/ checksums/
+    
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: gem-package
+        path: |
+          pkg/*.gem
+          checksums/*.sha512
+    
+    - name: Create GitHub Release
```

Comment:
move the "generate github release" to an own job...
publishing a release is nice to have. 
maybe run this in parallel to rubygems release, if this is state of the art.